spark/sql-20/src/main/scala/org/elasticsearch/spark/sql/DefaultSource.scala

```diff
@@ -156,9 +156,20 @@ private[sql] case class ElasticsearchRelation(parameters: Map[String, String], @
       }
     }
 
-    paramWithScan += (InternalConfigurationOptions.INTERNAL_ES_TARGET_FIELDS ->
-                      StringUtils.concatenate(filteredColumns.asInstanceOf[Array[Object]], StringUtils.DEFAULT_DELIMITER))
-
+    // Set fields to scroll over (_metadata is excluded, because it isn't a part of _source)
+    val scrollCSV = StringUtils.concatenate(filteredColumns.asInstanceOf[Array[Object]], StringUtils.DEFAULT_DELIMITER)
+    val requiredCSV = StringUtils.concatenate(requiredColumns.asInstanceOf[Array[Object]], StringUtils.DEFAULT_DELIMITER)
+    if (requiredCSV == cfg.getReadMetadataField)
+      // This is a small hack to avoid reading _source fields in case if only _metadata field is requested.
+      // By default, SearchRequestBuilder includes all fields, if INTERNAL_ES_TARGET_FIELDS is empty.
+      // To prevent us from querying useless data, we set _id field,
+      // which isn't a part of _source, but is allowed by SearchRequestBuilder
+      paramWithScan += (InternalConfigurationOptions.INTERNAL_ES_TARGET_FIELDS -> "_id")
```

Comment:
I'm not sure I like relying on this hack. Perhaps we could find a way to signal that we don't want to read the `_source` field at all by sending `_source=false` with the request. That way we can avoid depending on obscure and potentially easy-to-break functionality in Elasticsearch.